src/util.c

```diff
@@ -487,55 +489,87 @@ systemd_escape(char* string)
 gboolean
 netplan_delete_connection(const char* id, const char* rootdir)
 {
-    g_autofree gchar* del = NULL;
+    g_autofree gchar* yaml_path = NULL;
     g_autoptr(GError) error = NULL;
     NetplanNetDefinition* nd = NULL;
+    FILE* patch = NULL;
     gboolean ret = TRUE;
 
-    NetplanState* np_state = netplan_state_new();
-    NetplanParser* npp = netplan_parser_new();
+    NetplanParser* input_parser = netplan_parser_new();
+    NetplanState* input_state = netplan_state_new();
+    NetplanParser* output_parser = NULL;
+    NetplanState* output_state = NULL;
 
     /* parse all YAML files */
-    if (   !netplan_parser_load_yaml_hierarchy(npp, rootdir, &error)
-        || !netplan_state_import_parser_results(np_state, npp, &error)) {
+    if (   !netplan_parser_load_yaml_hierarchy(input_parser, rootdir, &error)
+        || !netplan_state_import_parser_results(input_state, input_parser, &error)) {
+        g_fprintf(stderr, "netplan_delete_connection: Cannot parse input: %s\n", error->message);
+        ret = FALSE;
+        goto cleanup;
+    }
+
+    /* find specified netdef in input state */
+    nd = netplan_state_get_netdef(input_state, id);
+    if (!nd) {
+        g_fprintf(stderr, "netplan_delete_connection: Cannot delete %s, does not exist.\n", id);
+        ret = FALSE;
+        goto cleanup;
+    }
+
+    /* Build up a tab-separated YAML path for this Netdef (e.g. network.ethernets.eth0=...) */
+    yaml_path = g_strdup_printf("network\t%s\t%s", netplan_def_type_name(nd->type), id);
+
+    /* create a temporary file in memory, to hold our YAML patch */
+    patch = tmpfile();
```

Comment:
issue: `tmpfile()` is disk-backed

I love the approach, but `tmpfile` can (and will, with glibc) create an actual file on the filesystem. You might want to have a look at `memfd_create` instead.

Reply:
Thanks, that's a good suggestion! The comment shows that it was the original intention, too :)

I had to bump `_X_OPEN_SOURCE=700` up to `_GNU_SOURCE` to enable the `memfd_*` functions, which are part of the GNU extensions and work around some conflicts with `__USE_MISC` (needed for `GLOB_BRACE` from glob.h)